src/StationAPIConnector.php

```diff
@@ -0,0 +1,250 @@
+        $body = $response->getBody()->getContents();
+
+        $full_data = json_decode($body, TRUE);
+
+        return $full_data;
+      }
+    }
+    catch (BadResponseException $e) {
+      $this->logger->get('tide_station_locator API response')
+        ->error('BadResponse error: @error', ['@error' => $e->getResponse()]);
+      return FALSE;
+    }
+
+    return [];
+  }
+
+  /**
+   * Helper method to get the Query Payload.
+   *
+   * @return array
+   *   Payload.
+   */
+  public function api_query_payload($delta = FALSE) {
+    // Get the 'last_api_access'.
+    $config = $this->configFactory->getEditable('tide_station_locator.settings');
+
+    if ($delta && !empty($config->get('last_api_access'))) {
+      $payload = [
+        'name' => 'STATION_LOCATION',
+        "productKey" => -1,
+        'attributes' => [
+          [
+            'name' => 'RECORD_MODDATE',
+            'value' => [$config->get('last_api_access')],
+            'operator' => 'gt',
+            'caseSensitive' => FALSE,
+          ],
+        ]
+      ];
+    }
+    else {
+      $payload = [
+        'name' => 'STATION_LOCATION',
+        'attributes' => [
+          [
+            'name' => 'ACTIVE',
+            'value' => ['Y'],
+            'caseSensitive' => FALSE,
+          ],
+        ],
+        'startIndex' => 1,
+        'recordCount' => 5,
+      ];
+    }
+
+    return $payload;
+  }
+
+  /**
+   * Helper method to get the 'last_access_api'.
+   *
+   * @return string
+   *   Config value last_api_access.
+   */
+  public function get_last_access_api() {
+    $config = $this->configFactory->getEditable('tide_station_locator.settings');
+    if (empty($config->get('last_api_access'))) {
+      $config->set('last_api_access', date('Y-m-d h:i:s'))->save();
+    }
+
+    return $config->get('last_api_access');
+  }
+
+  /**
+   * Helper method to format and save the JSON file.
+   *
+   * @param $stations
+   *   Records array.
+   *
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  public function get_formatted_station_response($stations) {
+    // Process the data so that we can save it in a format we need.
+    $final_stations = [];
+    foreach ($stations as $key => $station) {
+      // Loop through attributes.
+      foreach ($station['attributes'] as $attribute) {
+        // Add attribute name as the key.
+        $station[$attribute['name']] = $attribute['value'];
+      }
+      unset($station['attributes']);
+      $final_stations['records'][$key] = $station;
+    }
+
+    $file_save_path_stream_directory = 'public://';
+    $this->fileSystem->prepareDirectory($file_save_path_stream_directory, FileSystemInterface::CREATE_DIRECTORY | FileSystemInterface::MODIFY_PERMISSIONS);
+    $fileLocation = $file_save_path_stream_directory . '/' . 'stations.json';
+    // Save the processed JSON file.
+    $file = file_save_data(json_encode($final_stations), $fileLocation, FileSystemInterface::EXISTS_REPLACE);
+  }
```

Comment:
@g2410g We had talked about extracting the taxonomy terms etc. using this loop. Where is that code?

Reply:
Once that is done, we will not need the assets dir. We should change the paths for  state.csv/ accessibility.csv and specialityservicesandfacility.csv to `public://`